src/HornedBeasts.jsx

```diff
@@ -1,56 +1,36 @@
 import React from 'react';
-import { render } from 'react-dom';
 
 class HornedBeasts extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            clickedCount: 0
+        };
+    }
+
+    handleClickBeast = () => {
+        console.log("You've clicked on" + this.props.name);
+        this.setState({
+            clickedCount: this.state.clickedCount + 1
+        });
+    }
+
+    handleClickImage = (event) => {
+        this.props.handleOpenBeast(this.props.name, this.props.imgURL);
+    }
+
     render() {
         return (
             <div>
-                <img alt="uniwhal" src="http://3.bp.blogspot.com/_DBYF1AdFaHw/TE-f0cDQ24I/AAAAAAAACZg/l-FdTZ6M7z8/s1600/Unicorn_and_Narwhal_by_dinglehopper.jpg" />
-                <h2>{UniWhal}</h2>
-                <p>"A unicorn and narwhal touching horns</p>
-            </div>
-        )
-    };
-}
-class HornedBeasts extends React.Component {
-    render() {
-        return (
-            <div>
-                <h2>{Rhino}</h2>
-                <p>"Rhino family"</p>
-                <img alt="rhino" src="https://www.dhresource.com/0x0s/f2-albu-g5-M00-1A-11-rBVaI1hsIIiALxKzAAIHjSU3VkE490.jpg/wholesale-halloween-costume-prop-unicorn.jpg" />
+                <h2>{this.props.name}</h2>
+                <p onClick={this.handleClickBeast}>Clicked Beast</p>
+                <img
+                    src={this.props.imgURL}
+                    onClick={this.props.handleClickImage}
```

Comment:
handleClickImage isn't a prop so to use it you just need to change this to {this.handleClickImage}!